aiohttp/web_urldispatcher.py

```diff
@@ -674,6 +693,57 @@ def __repr__(self):
             prefix=self._prefix, app=self._app)
 
 
+class DynamicSubAppResource(DynamicResource):
+
+    def __init__(self, prefix, app):
+        super().__init__(prefix)
+        self._app = app
+        self._prefix = prefix
+
+    def url_for(self, *args, **kwargs):
```

Comment:
I see more complex problem.
`.url_for()` for all resources from sub-application are broken for dynamic prefixed resource.

Sorry, but the problem is blocker for merging.

Reply:
Yes, that's what I mentioned in the gitter channel.
I think rewriting `url_for()` to have a recursive architecture so that different types of resources could be chained require a lot of work, so I'd like to suggest it as a separate PR.
Maybe we could mark this PR as "provisional" if merged before it.

Reply:
I'm currently very busy due to my business work -- once I get some time after a few weeks, I'll give a look on the extension plan for `url_for()`.

Reply:
The problem is that without `url_for` fix not only `url_for()` for subapp itself doesn't work but all `url_for()` calls for subapp resources are broken.
That's why I think everything should be done in single PR.